tests/test_cli.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3
+
+from pytest import fixture, raises
+import subprocess
+from os import listdir
+import random
+
+
+@fixture
+def cli_args(shodan_key: str, censys_id: str, censys_secret: str) -> list:
+    """
+    Sets all needed arguments for execution and testing
+    :param shodan_key: shodan key value
+    :param censys_id: censys id value
+    :param censys_secret: censys secret value
+    :return: all constant arguments
+    """
+    sk = shodan_key
+    ci = censys_id
+    cs = censys_secret
+    base_args = [
+        "./grinder.py",
+        "-r",
+        "-u",
+        "-cu",
+        "-cp",
+        "-sk",
+        sk,
+        "-ci",
+        ci,
+        "-cs",
+        cs,
+        "-ni",
+    ]
+    return base_args
+
+
+@fixture
+def query_random() -> str:
+    """
+    Takes random query file from directory
+    :return: query file
+    """
+    queries = ["queries/" + query for query in listdir("queries")]
+    queries.remove("queries/test.json")
+    return random.choice(queries)
+
+
+def test_cli_base_args(cli_args: list, capfd) -> None:
+    """
+    Execute base arguments and check results
+    :param cli_args: needed arguments for execution and testing
+    :return: None
+    """
+    subprocess.run(["./grinder.py"])
+    assert "Usage: ./grinder.py -h for help" in str(capfd.readouterr().out)
+    subprocess.run(cli_args)
+    out, err = capfd.readouterr()
```

Comment:
Just a note: in this case you can use `check_output` method of `subprocess` module (without `readouterr`), but it's not the requirement - you just can check it by yourself, if it is more comfortable for you etc. If not, your solution is okay.

```python
In [39]: from subprocess import check_output
In [40]: check_output("uname", universal_newlines=True).split()
Out[40]: ['Darwin']
```